packages/build-utils/test/unit.traverse-up-directories.test.ts

```diff
@@ -0,0 +1,48 @@
+import { traverseUpDirectories } from '../src/fs/run-user-scripts';
```

Comment:
Do we walk to change this to `walk` instead of `traverse`?

We already have a similar `walkParentDirs()` function

Reply:
Actually, could this be an option to the existing `walkParentDirs()` instead of a new function?

Perhaps `{ all: true }`?

Reply:
Maybe, not sure. It might be confusing with `walkParentDirs()` (I think that function has a bad name for what it does FWIW). `traverse` might be a good distinction in this case since it's a generator.